internal/lm/nvml.go

```diff
@@ -218,6 +230,96 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(configFile string, device []resource.Device) (Labeler, error) {
+	if configFile == "" {
+		return nil, nil
+	}
+
+	// Read file and parse it
+	_, err := os.Stat(configFile)
+	if os.IsNotExist(err) {
+		return nil, nil
+	}
+	imexConfig, err := os.Open(configFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open imex config file: %v", err)
+	}
+	defer imexConfig.Close()
+
+	// check if the file is empty
+	stat, err := imexConfig.Stat()
+	if err != nil {
+		return nil, nil
+	}
+	if stat.Size() == 0 {
+		return nil, nil
+	}
```

Comment:
I don't think we need this explicit check. You can check after scanning `if len(ips) == 0`.

Reply:
suggestion taken